### PR TITLE
{Network} `az network express-route`: Migrate to atomic commands

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

no migrate command: 
az network express-route list-route-tables-summary
("as" conflicts)
migrate command：
az network express-route
az network express-route auth
az network express-route gateway
az network express-route gateway connection
az network express-route peering
az network express-route peering connection
az network express-route peering peer-connection
az network express-route port
az network express-route port link
az network express-route port location
**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -32,6 +32,12 @@
 import tempfile
 import requests
 
+from .aaz.latest.network.express_route import Create as _ExpressRouteCreate, Update as _ExpressRouteUpdate
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:35]: Please move this code block to "from .aaz.latest.network.application_gateway._update import Update as _ApplicationGatewayUpdate" and keep lexicographical order.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 2]
```diff
@@ -32,6 +32,12 @@
...
+from .aaz.latest.network.express_route.gateway.connection import Create as _ExpressRouteConnectionCreate, \
+    Update as _ExpressRouteConnectionUpdate
+from .aaz.latest.network.express_route.peering import Create as _ExpressRoutePeeringCreate, \
+    Update as _ExpressRoutePeeringUpdate
+
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:40]: Please move this code block to "from .aaz.latest.network.application_gateway._update import Update as _ApplicationGatewayUpdate" and keep the lexicographical order.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -32,6 +32,12 @@
 import tempfile
 import requests
 
+from .aaz.latest.network.express_route import Create as _ExpressRouteCreate, Update as _ExpressRouteUpdate
+from .aaz.latest.network.express_route.gateway.connection import Create as _ExpressRouteConnectionCreate, \
+    Update as _ExpressRouteConnectionUpdate
+from .aaz.latest.network.express_route.peering import Create as _ExpressRoutePeeringCreate, \
+    Update as _ExpressRoutePeeringUpdate
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:39]: Please move this code block to "from .aaz.latest.network.application_gateway._update import Update as _ApplicationGatewayUpdate" and keep the lexicographical order.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 4]
```diff
@@ -3094,64 +3090,62 @@ def lists_match(l1, l2):
...
+            options=["--bandwidth"],
+            help="Bandwidth of the circuit. Usage: INT {Mbps,Gbps}. Defaults to Mbps.",
+            nullable=True,
+        )
+        return args_schema
 
+    def _cli_arguments_loader(self):
+        args = super()._cli_arguments_loader()
+        args = [(name, arg) for (name, arg) in args if name not in ("bandwidth_in_mbps", "bandwidth_of_circuit")]
+        return args
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3108]: Please leverage "args_schema.\<arg-name\>._registered = False" in "_build_arguments_schema".

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 5]
```diff
@@ -3094,64 +3090,62 @@ def lists_match(l1, l2):
...
-                         allow_global_reach=None, express_route_port=None,
-                         allow_classic_operations=None):
+    def pre_operations(self):
+        args = self.ctx.args
+        args.sku_name = '{}_{}'.format(args.sku_tier, args.sku_family)
+        args.bandwidth_of_circuit = (args.bandwidth.to_serialized_data() / 1000.0) if args.express_route_port else None
+        args.bandwidth_in_mbps = (args.bandwidth) if not args.express_route_port else None
+        if args.express_route_port:
+            args.provider = None
+            args.peering_location = None
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3117]: Please add "if has_value(args.\<arg-name\>)" for each argument, it represents whether the user input the argument.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 6]
```diff
@@ -3278,6 +3272,58 @@ def create_express_route_peering(
...
+
+        return args_schema
+
+    def _cli_arguments_loader(self):
+        args = super()._cli_arguments_loader()
+        args = [(name, arg) for (name, arg) in args if name not in ("ipv6_peering_config", "peering_name")]
+        return args
+
+    def pre_operations(self):
+        from azure.cli.core.aaz import AAZUndefined
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3295]: What is it for?

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 7]
```diff
@@ -3278,6 +3272,58 @@ def create_express_route_peering(
...
+        args = self.ctx.args
+        args.peering_name = args.peering_type
+        if args.ip_version == 'IPv6':
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3298]: Similar as above.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 8]
```diff
@@ -3308,6 +3354,53 @@ def _create_or_update_ipv6_peering(cmd, config, primary_peer_address_prefix, sec
...
+            help="The IP version to update Microsoft Peering settings for. Allowed values: IPv4, IPv6.  Default: IPv4.",
+            default='IPv4'
+        )
+
+        return args_schema
+
+    def _cli_arguments_loader(self):
+        args = super()._cli_arguments_loader()
+        args = [(name, arg) for (name, arg) in args if name not in ("ipv6_peering_config", "peering_type", "connections")]
+        return args
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3375]: Similar as above.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 9]
```diff
@@ -3308,6 +3354,53 @@ def _create_or_update_ipv6_peering(cmd, config, primary_peer_address_prefix, sec
...
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.ip_version = AAZStrArg(
+            options=['--ip-version'],
+            arg_group="Microsoft Peering",
+            help="The IP version to update Microsoft Peering settings for. Allowed values: IPv4, IPv6.  Default: IPv4.",
+            default='IPv4'
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3367]: Missing "nullable=True" as it's a update command.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 10]
```diff
@@ -18,12 +18,19 @@
     UnrecognizedArgumentError, ResourceNotFoundError, CLIInternalError, ArgumentUsageError, \
     MutuallyExclusiveArgumentError
 from azure.cli.core.profiles import ResourceType, supported_api_version
+from azure.cli.core.aaz import has_value
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:21]: "import" should in lexicographical order, please move it to line 12.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 11]
```diff
@@ -32,6 +39,8 @@
 import tempfile
 import requests
 
+
+
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:43]: Remove these blank lines.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 12]
```diff
@@ -3387,16 +3400,33 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
+            )
+        )
         args_schema.associated_id._registered = False
         args_schema.propagated_ids._registered = False
         return args_schema
 
     def pre_operations(self):
         args = self.ctx.args
+        if has_value(args.peering) and '/' not in args.peering.to_serialized_data():
+            usage_error = CLIError('usage error: --peering ID | --peering NAME --circuit-name CIRCUIT')
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3418]: Please refine it by https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md.

File: src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/express_route/gateway/connection/_update.py [hunk 13]
```diff
@@ -81,14 +81,11 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
         _args_schema = cls._args_schema
-        _args_schema.peering = AAZResourceIdArg(
+        _args_schema.peering = AAZStrArg(
             options=["--peering"],
             arg_group="Peering",
             help="Name or ID of an ExpressRoute peering.",
-            nullable=True,
-            fmt=AAZResourceIdArgFormat(
-                template="/subscriptions/{subscription}/resourceGroups/{resource_group}/providers/Microsoft.Network/expressRouteCircuits/{circuit_name}/peerings/{}"
-            )
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/express_route/gateway/connection/_update.py]: Better to move such logic to custom.py, i.e., args_schema.peering._fmt=xxxx.

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 14]
```diff
@@ -3094,86 +3091,65 @@ def lists_match(l1, l2):
...
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZListArg, AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.bandwidth = AAZListArg(
+            options=["--bandwidth"],
+            help="Bandwidth of the circuit. Usage: INT {Mbps,Gbps}. Defaults to Mbps.",
+            nullable=True,
+        )
+        args_schema.bandwidth.Element = AAZStrArg()
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:3133]: Missing "nullable".

**Review by necusjz — APPROVED** [review]

**Pull request merged**